### PR TITLE
Python: make imports lazy, add .sql as an alias for .query, and add integration functions with polars

### PR DESCRIPTION
**Pull request by Mytherin**
This PR makes several improvements to the Python module:

* All imports are made lazy. This greatly speeds up the import time of the module.
* Adds the `.sql` method as an alias for `.query`, i.e. this now works:

```py
> import duckdb
> duckdb.sql('select 42')
┌───────┐
│  42   │
│ int32 │
├───────┤
│    42 │
└───────┘
```

* Adds the `.pl()` function to convert results of queries to Polars DataFrames, i.e. this now works:

```py
> import duckdb
> duckdb.sql('select 42').pl()
shape: (1, 1)
┌─────┐
│ 42  │
│ --- │
│ i32 │
╞═════╡
│ 42  │
└─────┘
```

* Adds support for replacement scans over Polars DataFrames, i.e. this now works:
```py
import duckdb
import polars
df = pl.DataFrame._from_dict({'a': 42})
duckdb.sql('select * from df').pl()
┌─────┐
│ a   │
│ --- │
│ i64 │
╞═════╡
│ 42  │
└─────┘
```

### CODE REVIEW

**Comment by Tishj** [comment]
Cool, awesome functionality.

I think at the time I added the import cache we tried to make it lazily loaded but couldn't get it to work (iirc)
Definitely a huge improvement!

File: tools/pythonpkg/tests/fast/arrow/test_polars.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import duckdb
+import pytest
+
+class TestPolars(object):
+    def test_polars(self,duckdb_cursor):
+        pd = pytest.importorskip("pandas")
```
Comment (ritchie46) [inline tools/pythonpkg/tests/fast/arrow/test_polars.py:6]: Not really needed to go back and forth via pandas here.

```python
df = pl.DataFrame(
    {
        "A": [1, 2, 3, 4, 5],
        "fruits": ["banana", "banana", "apple", "apple", "banana"],
        "B": [5, 4, 3, 2, 1],
        "cars": ["beetle", "audi", "beetle", "beetle", "beetle"],
    }
)
# scan plus return a polars dataframe
polars_result = duckdb.sql('SELECT * FROM df').pl()
pl.testing.assert_frame_equal(df, polars_result)

# now do the same for a lazy dataframe
lazy_df = df.lazy()
lazy_result = duckdb.sql('SELECT * FROM lazy_df').pl()
pl.testing.assert_frame_equal(df, lazy_result)

con = duckdb.connect()
con_result = con.execute('SELECT * FROM df').pl()
pl.testing.assert_frame_equal(df, con_result)
```

Comment (Mytherin) [reply tools/pythonpkg/tests/fast/arrow/test_polars.py:6]: Hm - that does not seem to work for me.

```py
__________________________________________________________________________________ TestPolars.test_polars ___________________________________________________________________________________

self = <test_polars.TestPolars object at 0x104b60b80>, duckdb_cursor = <duckdb.DuckDBPyConnection object at 0x104bd2df0>

    def test_polars(self,duckdb_cursor):
        pl = pytest.importorskip("polars")
        df = pl.DataFrame(
            {
                "A": [1, 2, 3, 4, 5],
                "fruits": ["banana", "banana", "apple", "apple", "banana"],
                "B": [5, 4, 3, 2, 1],
                "cars": ["beetle", "audi", "beetle", "beetle", "beetle"],
            }
        )
        # scan plus return a polars dataframe
        polars_result = duckdb.sql('SELECT * FROM df').pl()
>       pl.testing.assert_frame_equal(df, polars_result)
E       AttributeError: module 'polars' has no attribute 'testing'
```

Perhaps I need to install additional polars modules for that?

Comment (ritchie46) [reply tools/pythonpkg/tests/fast/arrow/test_polars.py:6]: My bad, we need to import it explicitly (that keeps import times in check):

```python
import polars as pl
import polars.testing

```

Comment (Mytherin) [reply tools/pythonpkg/tests/fast/arrow/test_polars.py]: That makes sense - thanks. I have updated the test to use Polars' native functions.

**Pull request merged**